File: 1975-maximum-matrix-sum/1975-maximum-matrix-sum.cpp
class Solution {
public:
    long long maxMatrixSum(vector<vector<int>>& matrix) {
        long long sum = 0;
        int mn = 1e9,cnt = 0;
        for(auto row : matrix){
            for(auto x : row){
                sum+=abs(x);
                mn = min(mn,abs(x));
                cnt+=(x<0);
            }
        }
        return sum - (cnt&1 ? 2*mn : 0);
    }
};

File: 1975-maximum-matrix-sum/1975-maximum-matrix-sum.py
class Solution:
    def maxMatrixSum(self, matrix):
        total_sum = 0
        min_value = float('inf')
        negative_count = 0

        for row in matrix:
            for x in row:
                total_sum += abs(x)
                min_value = min(min_value, abs(x))
                if x < 0:
                    negative_count += 1

        return total_sum - (2 * min_value if negative_count % 2 != 0 else 0)
